base/tuple.jl

```diff
@@ -148,19 +148,22 @@ function map(f, t::Tuple, s::Tuple)
 end
 function map(f, t::Any16, s::Any16)
     n = length(t)
-    A = Array{Any}(n)
+    A = Array{Any,1}(n)
     for i = 1:n
         A[i] = f(t[i], s[i])
     end
     (A...,)
 end
 # n argument function
 heads() = ()
-heads(t::Tuple, ts::Tuple...) = (t[1], heads(ts...)...)
+heads(t::Tuple, ts::Tuple...) = (@_inline_meta; (t[1], heads(ts...)...))
 tails() = ()
-tails(t::Tuple, ts::Tuple...) = (tail(t), tails(ts...)...)
-map(f, ::Tuple{}, ts::Tuple...) = ()
-map(f, t1::Tuple, t2::Tuple, ts::Tuple...) = (f(heads(t1, t2, ts...)...), map(f, tails(t1, t2, ts...)...)...)
+tails(t::Tuple, ts::Tuple...) = (@_inline_meta; (tail(t), tails(ts...)...))
+map(f, ::Tuple{}...) = ()
+map(f, t::Tuple, ts::Tuple...) = (@_inline_meta;
```

Comment:
Yes. I see the first definition exceeds the limit by a few characters though the second does not; I assume you split the second definition for consistency, so cheers :).